packages/components/src/components/ResizableBox/ResizableBox.tsx

```diff
@@ -179,30 +196,67 @@ const getMaxResult = (max: number | undefined, result: number) =>
 const isInDisabledInterval = (value: number, interval?: DisabledInterval) =>
     interval && value > interval[0] && value < interval[1];
 
-const calculateDisabledHeightInterval = (
-    result: number,
-    disabledHeightInterval?: DisabledInterval,
-) => {
-    if (disabledHeightInterval && isInDisabledInterval(result, disabledHeightInterval)) {
-        return result < (disabledHeightInterval[0] + disabledHeightInterval[1]) / 2
-            ? disabledHeightInterval[0]
-            : disabledHeightInterval[1];
+const calculateDisabledInterval = (result: number, disabledInterval?: DisabledInterval) => {
+    if (disabledInterval && isInDisabledInterval(result, disabledInterval)) {
+        return result < (disabledInterval[0] + disabledInterval[1]) / 2
+            ? disabledInterval[0]
+            : disabledInterval[1];
     }
 
     return result;
 };
 
-const calculateDisabledWidthInterval = (
-    result: number,
-    disabledWidthInterval?: DisabledInterval,
-) => {
-    if (disabledWidthInterval && isInDisabledInterval(result, disabledWidthInterval)) {
-        return result < (disabledWidthInterval[0] + disabledWidthInterval[1]) / 2
-            ? disabledWidthInterval[0]
-            : disabledWidthInterval[1];
-    }
+type ResizeState = {
+    x: number;
+    y: number;
+    width: number;
+    height: number;
+    isResizing: boolean;
+    isHovering: boolean;
+    direction: Direction | null;
+};
 
-    return result;
+type ResizeAction =
+    | { type: 'SET_POSITION'; x: number; y: number }
+    | { type: 'SET_WIDTH'; width: number }
+    | { type: 'SET_HEIGHT'; height: number }
+    | { type: 'START_RESIZE'; direction: Direction }
+    | { type: 'STOP_RESIZE' }
+    | { type: 'MOUSE_OVER'; direction: Direction }
+    | { type: 'MOUSE_OUT' };
+
+const resizeReducer = (state: ResizeState, action: ResizeAction): ResizeState => {
+    switch (action.type) {
+        case 'SET_POSITION':
+            return { ...state, x: action.x, y: action.y };
+        case 'SET_WIDTH':
+            return { ...state, width: action.width };
+        case 'SET_HEIGHT':
+            return { ...state, height: action.height };
+        case 'START_RESIZE':
+            return {
+                ...state,
+                isResizing: true,
+                isHovering: false,
+                direction: action.direction,
+            };
+        case 'STOP_RESIZE':
+            return { ...state, isResizing: false };
+        case 'MOUSE_OVER':
+            return {
+                ...state,
+                isHovering: true,
+                direction: action.direction,
+            };
+        case 'MOUSE_OUT':
+            return {
+                ...state,
+                isHovering: false,
+                direction: state.isResizing ? state.direction : null,
+            };
+        default:
+            return state;
+    }
 };
 
 export const ResizableBox = ({
```

Comment:
Good job here, this component needed some love. Reducer and 100% values looks much better than before